src/game/lara/lara_cheat.c

```diff
@@ -269,6 +339,12 @@ bool Lara_Cheat_Teleport(int32_t x, int32_t y, int32_t z)
         g_LaraItem->goal_anim_state = LS_STOP;
         g_LaraItem->anim_num = g_Objects[O_LARA].anim_idx + LA_STAND_STILL;
         g_LaraItem->frame_num = g_Anims[g_LaraItem->anim_num].frame_base;
+        g_LaraItem->rot.x = 0;
+        g_LaraItem->rot.z = 0;
+        g_Lara.head_x_rot = 0;
+        g_Lara.head_y_rot = 0;
+        g_Lara.torso_x_rot = 0;
+        g_Lara.torso_y_rot = 0;
```

Comment:
I think we can actually delete lines 323 to 348 altogether. `Lara_Control` takes care of switching to/from water - it's more complex than TR1 because of the treading abilities, so the issue itself is in testing water heights there after we've already changed her status here. I've tried removing these lines and it behaves well, she does the splash animation when tp'ing into water, and the fall anim when tp'ing to land. This behaves the same if you tp to a dry room then `/flip` to a water room e.g. Rig room 36, so it'd also be consistent with that.

Reply:
Unfortunately removing this causes Lara to die in Offshore Rig when trying to teleport to room 9.

Reply:
Interestingly, I had this too, but I forced a full recompile and it seems to resolve that?

Reply:
Actually no, it still happens in room 9.
That's really odd.
Not a big deal, this way works of course so probably best for now.